packages/app-store/routing-forms/api/responses/[formId].ts

```diff
@@ -63,15 +66,34 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     throw new Error("formId must be provided");
   }
 
+  const session = await getSession({ req });
+
+  if (!session) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+
   const form = await prisma.app_RoutingForms_Form.findFirst({
     where: {
       id: formId,
+      ...entityPrismaWhereClause({ userId: session.user.id }),
```

Comment:
`entityPrismaWhereClause` ensures that everywhere we are querying only those forms which user has access to.